### PR TITLE
Skip cockpit-test-assets when installing in a target machine

### PR DESCRIPTION
**Pull request by stefwalter**
When vm-install is used with --address, these machines are used for demoing, or debugging, and we don't want all the playground stuff cluttering up the sidebar.

### CODE REVIEW

File: test/common/vmmanage.py [hunk 1]
```diff
@@ -97,9 +97,12 @@ def build_and_install(install_image, build_image, args):
     args.setdefault("containers", False)
     args.setdefault("address", None)
     try:
-        skip = "cockpit-ostree"
-        if install_image and "fedora-atomic" in install_image:
-            skip = None
+        skips = [ ]
+        if install_image and "fedora-atomic" not in install_image:
```
Comment (dperpeet) [inline test/common/vmmanage.py]: Why just fedora-atomic? This seems to be a relic - I suggest just `atomic` to also cover RHEL Atomic Host.

**Comment by dperpeet** [comment]
I'm fine with this in general, but I'd like to see an option to force installation of test-assets nevertheless. Maybe `-t`?

File: test/common/vmmanage.py (hunk 1 above)
Comment (stefwalter) [inline test/common/vmmanage.py]: That happens in another pull request https://github.com/cockpit-project/cockpit/pull/4047

**Comment by stefwalter** [comment]
> 

I'm fine with this in general, but I'd like to see an option to force installation of test-assets nevertheless. Maybe -t?

It seems unnecessary. This only applies when --address is used. Once someone wants to use --address with cockpit-test-assets ... makes sense to add that option but until then i think the YAGNI principle should apply.

**Comment by dperpeet** [comment]
> It seems unnecessary. This only applies when --address is used. Once someone wants to use --address with cockpit-test-assets ... makes sense to add that option but until then i think the YAGNI principle should apply.

ok, makes sense

**Comment by dperpeet** [comment]
I think the way these "skip" packages are passed is flawed.
[atomic.install](https://github.com/cockpit-project/cockpit/blob/09033594791d284976957d72b6c64e61e69aa0e6/test/images/scripts/lib/atomic.install#L244) receives them as a list, so [vmmanage](https://github.com/cockpit-project/cockpit/blob/805d1d6987c4f92ac8a0825c36cebd55a9bd0bed/test/common/vmmanage.py#L43) should handle the list case differently. As it stands, I believe the [startswith logic](https://github.com/cockpit-project/cockpit/blob/09033594791d284976957d72b6c64e61e69aa0e6/test/images/scripts/lib/atomic.install#L256) in atomic.install will fail if more than one package is skipped, e.g. `test-assets` and `ostree`.

**Comment by dperpeet** [comment]
`atomic.install` expects to be called like `--skip 'cockpit-ostree' --skip 'cockpit-test-assets'`

**Comment by dperpeet** [comment]
needs a rebase and the above mentioned issues need an answer

**Comment by stefwalter** [comment]
Fixed up.

**Comment by stefwalter** [comment]
Tested manually against a running machine.

File: test/common/vmmanage.py [hunk 2]
```diff
@@ -66,7 +67,7 @@ def build_and_maybe_install(image, do_install=False, skip=None, args=None):
         finally:
             machine.stop()
 
-def only_install(image, skip=None, args=None, address=None):
+def only_install(image, skips=[], args=None, address=None):
```
Comment (dperpeet) [inline test/common/vmmanage.py]: Mutable types as a default argument for a kwarg are usually a bad idea. Better to use None and instantiate to a blank value in the code.

File: test/common/vmmanage.py [hunk 3]
```diff
@@ -30,22 +30,23 @@ def upload_scripts(machine, args):
...
                                                          " --build" if do_build else "",
                                                          " --install" if do_install else "",
-                                                         " --skip '%s'" % skip if skip else "",
+                                                         " ".join(skip_args),
                                                          " '%s'" % arg if arg else ""))
     if install and args["containers"]:
         machine.execute("/var/lib/testvm/containers.install")
 
-def build_and_maybe_install(image, do_install=False, skip=None, args=None):
+def build_and_maybe_install(image, do_install=False, skips=[], args=None):
```
Comment (dperpeet) [inline test/common/vmmanage.py]: Mutable types as a default argument for a kwarg are usually a bad idea. Better to use None and instantiate to a blank value in the code.

File: test/common/vmmanage.py (hunk 2 above)
Comment (stefwalter) [inline test/common/vmmanage.py]: Done.

File: test/common/vmmanage.py (hunk 3 above)
Comment (stefwalter) [inline test/common/vmmanage.py]: Done.

**Pull request closed without merging**